packages/cli/src/utils/reporter.ts

```diff
@@ -0,0 +1,157 @@
+import { saveToFileSync } from './filesystem';
```

Comment:
nitpick (non-blocking):

Personally, I’d avoid placing the reporter under utils, since it’s more of a core component/service than a util or helper (similar with Logger). Following DDD principles, I'd rather put it in its own `src/reporter` folder, or `src/lib/reporter`